crates/forge/tests/integration/gas.rs

```diff
@@ -5,6 +5,64 @@ use test_utils::running_tests::run_test_case;
 use test_utils::{assert_gas, assert_passed, test_case};
 
 // gas values comes from https://book.starknet.io/ch03-01-02-fee-mechanism.html#computation
+#[test]
+fn test_declare_cost_is_omitted() {
+    let test = test_case!(
+        indoc!(
+            r"
+            use snforge_std::declare;
+
+            #[test]
+            fn test_declare() {
+                declare('GasChecker');
+            }
+        "
+        ),
+        Contract::from_code_path(
+            "GasChecker".to_string(),
+            Path::new("tests/data/contracts/gas_checker.cairo"),
+        )
+        .unwrap()
+    );
+
+    let result = run_test_case(&test);
+
+    assert_passed!(result);
+    // 1 - initial gas cost (consists of steps required to run the code)
+    assert_gas!(result, "test_declare", 1);
+}
+
+#[test]
+fn test_deploy_syscall_cost() {
+    let test = test_case!(
+        indoc!(
+            r"
+            use snforge_std::declare;
+            use starknet::{SyscallResult, deploy_syscall};
+
+            #[test]
+            fn test_deploy_syscall() {
+                let contract = declare('GasChecker');
+                deploy_syscall(contract.class_hash, 0, array![].span(), false);
+            }
+        "
+        ),
+        Contract::from_code_path(
+            "GasChecker".to_string(),
+            Path::new("tests/data/contracts/gas_checker.cairo"),
+        )
+        .unwrap()
+    );
+
+    let result = run_test_case(&test);
+
+    assert_passed!(result);
+    // 1224 = 2 * cost per 32-byte word (contract_address and contract modification info)
+    // 612 - updated class_hash (through deploy)
```

Comment:
We should investigate that since it is inconsistent with docs, worth asking on starkware-swm or sth